src/Microsoft.Diagnostics.ExtensionCommands/NativeAddressHelper.cs

```diff
@@ -0,0 +1,356 @@
+﻿using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System;
+using Microsoft.Diagnostics.Runtime.DataReaders.Implementation;
+using System.Buffers;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    internal sealed class NativeAddressHelper
+    {
+        private readonly IDataReader _reader;
+        private readonly ClrRuntime[] _runtimes;
+        private readonly IMemoryRegionService _memory;
+        private readonly IModuleService _modules;
+
+        public NativeAddressHelper(IDataReader reader, ClrRuntime[] runtimes, IMemoryRegionService memory, IModuleService modules)
+        {
+            _reader = reader;
```

Comment:
Ah good feedback, this was actually a correctness issue too.  You don't implement IThreadData (you only consume it), so that actually broke some of the code here.  I have a fix I will push shortly.